### ISSUE TITLE
When to support Linux deployment

### ISSUE THREAD

**Issue by haiyang1024** [StunlockStudios/vrising-dedicated-server-instructions#97]
When to support Linux deployment

**Comment by d-allin** [comment]
Runs fine with Wine, dockerized. Plenty of images out there. Running this one fine.
https://github.com/TrueOsiris/docker-vrising

**Comment by paulvilla** [comment]
> Runs fine with Wine, dockerized. Plenty of images out there. Running this one fine. https://github.com/TrueOsiris/docker-vrising

wine is a bad patch, I tell you because I have created many servers and it is not the solution, when you have or are hosting more than one server, the consumption and possible failures that would not occur in the native system will be said enormously

Stop announcing that option as if it were a gift from God, because the only thing it is used for is to get over the pothole, but if what you want is stability, you have to rule it out completely.

**Comment by d-allin** [comment]
> Stop advertising that option as if it were a gift from God

All I said was that it was working. Yes, obviously it's a temporary patch/solution until the developers release their linux version. I hope it can be done soon.

I share your concerns with Wine. If you know of any alternatives while we wait, I'd genuinely like to know of them so I can test those deployments.

**Comment by ubergarm** [comment]
I'm using `wine` for now too on Ubuntu 22.04 LTS, looking forward to native linux binaries too though.

**Comment by Umlatt** [comment]
Agree. When self-hosting in the cloud, the wine addition takes a ~350MB image to a ~2GB image. Lot's of overhead to run it with wine

**Comment by JamieeLee** [comment]
> Agree. When self-hosting in the cloud, the wine addition takes a ~350MB image to a ~2GB image. Lot's of overhead to run it with wine

Actually the wine binary overhead is actually non existent  for library files , honestly its the one thing about it that is ok. But the inflated  space requirement because you need all the workarounds agree with. I tested the ram between a native  and wine version ram usage and cpu is about the same.   However there is a bigger risk that an update to the windows server breaks wine. Then in turn breaks your server.

**Comment by Umlatt** [comment]
> > Agree. When self-hosting in the cloud, the wine addition takes a ~350MB image to a ~2GB image. Lot's of overhead to run it with wine
> 
> Actually the wine binary overhead is actually non existent for library files , honestly its the one thing about it that is ok. But the inflated space requirement because you need all the workarounds agree with. I tested the ram between a native and wine version ram usage and cpu is about the same. However there is a bigger risk that an update to the windows server breaks wine. Then in turn breaks your server.
> 
> TLDR: We need a linux server binary like yesterday.

You're quite correct. I was maybe not specific enough, was referring to storage overhead. 
I can't comment, personally, on the actual compute overhead, as I have only run it using the wine workarounds. 

I need to keep the images as tight and tidy as possible, as the cloud provider I use charges per GB /hour, and I host several different game servers for different use cases.

Speaking of which, just an FYI, the autosaves can add up (capacity wise), if you don't clean them out.
Have about 130 saves (defaults) and the directory is sitting at about 5GB utilisation.

**Comment by RuudschMaHinda** [comment]
So, the question is "When can we natively host on linux servers?"
And the answers are "Run a runtime environment."

That does not answer the question. Answers could be:
"It's not planned, maybe look into wine solutions."
"It is planned, but we don't know yet."
"It is planned, currently we are looking at [DATE]."
I probably forgot possible answers. But "just use wine" is not an answer to this question. It's like asking "When can I get my car?" and getting "Just use the bus" as a reply. It does not answer the question.

So, wine: Open a different Issue for that.
Linux: Stay on this issue and drop the wine discussion.
